Frontend/Camera2Basic/Camera2BasicFragment.cs

```diff
@@ -0,0 +1,749 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive.Concurrency;
+using Android;
+using Android.App;
+using Android.Content;
+using Android.Content.PM;
+using Android.OS;
+using Android.Util;
+using Android.Views;
+using Android.Widget;
+using Android.Hardware.Camera2;
+using Android.Graphics;
+using Android.Hardware.Camera2.Params;
+using Android.Media;
+using Android.Support.V13.App;
+using Android.Support.V4.Content;
+using Camera2Basic.Listeners;
+using Java.IO;
+using Java.Lang;
+using Java.Util;
+using Java.Util.Concurrent;
+using Boolean = Java.Lang.Boolean;
+using Math = Java.Lang.Math;
+using Observable = System.Reactive.Linq.Observable;
+using Orientation = Android.Content.Res.Orientation;
+
+namespace Camera2Basic
+{
+    public class Camera2BasicFragment : Fragment, View.IOnClickListener, FragmentCompat.IOnRequestPermissionsResultCallback
+    {
+        private static readonly SparseIntArray ORIENTATIONS = new SparseIntArray();
```

Comment:
why readonly instead of const, if you don't assign it in the constructor? also, what about moving some of constants into config file?